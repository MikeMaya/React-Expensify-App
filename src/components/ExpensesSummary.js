import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <h1>
            Viewing {props.expensesCount} 
            {props.expensesCount === 1 ? ' expense ':' expenses '}
            totalling {numeral(props.expensesTotal / 100).format("$0,0.00")}
            
        </h1>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses= selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);