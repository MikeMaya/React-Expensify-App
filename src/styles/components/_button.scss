.button-login {
    background: $blue;
    border: none;   
    color: white;
    font-weight: 300;
    font-size: $font-size-large;
    margin-bottom: $s-size;
    padding: $s-size;
}