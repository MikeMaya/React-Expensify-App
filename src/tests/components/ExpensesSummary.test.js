import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses'

test('Render Expenses Summary - Single Expense', () => {
    const wrapper = shallow(
        <ExpensesSummary 
            expensesCount={1} 
            expensesTotal={9568}
        />);
    expect(wrapper).toMatchSnapshot();
});

test('Render Expenses Summary - Multiple Expenses', () => {
    const wrapper = shallow(
        <ExpensesSummary 
            expensesCount={2} 
            expensesTotal={9568}
        />);
    expect(wrapper).toMatchSnapshot();
});