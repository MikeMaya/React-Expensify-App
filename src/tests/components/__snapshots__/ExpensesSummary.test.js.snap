// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Expenses Summary - Multiple Expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     expenses 
    totalling 
    $95.68
  </h1>
</div>
`;

exports[`Render Expenses Summary - Single Expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense 
    totalling 
    $95.68
  </h1>
</div>
`;
